Animation.cs

```diff
@@ -102,6 +139,18 @@ public void NextFrame()
 
         currentFrameId = (currentFrameId + 1) % Frames.Count;
         currentDelay = Frames[currentFrameId].Delay.Milliseconds;
-        CurrentFrame = Frames[currentFrameId].Texture;
+        
+        if (image == null)
+        {
+            // If the class is allocated from the generic constructor, then image is null and we should use Animation.Frame.
+            CurrentFrame = Frames[currentFrameId].Texture;
+        }
+        else
+        {
+            // Otherwise we have the source image and would be better off saving memory.
+            CurrentFrame.Dispose();
+            CurrentFrame = AssetLoader.TextureFromImage(image.Frames.CloneFrame(currentFrameId));
```

Comment:
`image.Frames.CloneFrame` returns an `Image` which is `IDisposable`. In other words, with this approach you need to dispose not only the `Texture2D`, but also the previous frame's `Image` (created by `CloneFrame`) to avoid memory leaks. .NET can dispose of it automatically but IIRC it is not guaranteed.

You might be able to dispose it right away here after the `TextureFromImage` call.

Reply:
Done by d14fd1b361190d70d6f6e3ff84d7962b0e7771ec.